olive/engine/footprint.py

```diff
@@ -0,0 +1,178 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from collections import OrderedDict
+from datetime import datetime
+from pathlib import Path
+
+from olive.common.config_utils import ConfigBase, config_json_dumps, config_json_loads
+
+logger = logging.getLogger(__name__)
+
+
+class FootprintNodeMetric(ConfigBase):
+    value: dict
```

Comment:
>dict [](http://example.com/codeflow?start=11&length=4)

for these dict annotation, would you please help specify the key and value type?

Reply:
Same structure with the results of metrics evaluation. {metrics_name: value, ....}. 
Will tried to add EPs/accelerators into account when Mike's accelerators changes is merged.

Reply:
Also, already add the comments in code for readability.